docs/cluster-agent/HORIZONTAL_POD_AUTOSCALING.md

```diff
@@ -0,0 +1,262 @@
+
+Every 30 seconds (this can be configured) Kubernetes will query the DCA to get the value of this metric and will autoscale proportionally if necessary.
+For advanced use cases, it is possible to have several metrics in the same HPA, as you can see [here](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#support-for-multiple-metrics) the largest of the proposed value will be the one chosen.
+
+Now, let's create our nginx deployment:
+
+`kubectl apply -f manifests/cluster-agent/hpa-example/nginx.yaml`
+
+Then, let's apply the HPA
+
+`kubectl apply -f manifests/cluster-agent/hpa-example/hpa-manifest.yaml`
+
+### Stressing your service
+
+Now is time to see the magic happening!
+
+If you curl the IP of the nginx service as follows:
+`curl <nginx_svc>:8090/nginx_status`, you should receive an output like:
+
+```
+$ curl 192.168.254.216:8090/nginx_status
+
+Active connections: 1 
+server accepts handled requests
+ 1 1 1 
+Reading: 0 Writing: 1 Waiting: 0 
+```
+
+Behind the scene, the number of request per second also increased. 
+This metric is being collected by the node agent, as it autodiscovered the nginx pod through its annotations (for more information on how autodiscovery works, see our doc [here](https://docs.datadoghq.com/agent/autodiscovery/#template-source-kubernetes-pod-annotations)).
+Therefore, if you stress it, you will see the uptick in your Datadog app. 
+As you referenced this metric in your HPA manifest, the DCA is also pulling its latest value every 20 seconds.
+Then, as Kubernetes queries the DCA to get this value, it will notice that the number is going above the threshold and will autoscale accordingly.
+
+Let's do it!
+
+Just run `while true; do curl <nginx_svc>:8090/nginx_status; sleep 0.1; done`
+And you should soon see the number of requests per second spiking, going above 9 which is the threshold over which we want to autoscale out nginx boxes.
+Then, you should see new nginx pods being created:
+
+```
+PODS:
+
+NAMESPACE     NAME                                     READY     STATUS    RESTARTS   AGE
+default       datadog-cluster-agent-7b7f6d5547-cmdtc   1/1       Running   0          9m
+default       nginx-6757dd8769-5xzp2                   1/1       Running   0          2m
+default       nginx-6757dd8769-k6h6x                   1/1       Running   0          2m
+default       nginx-6757dd8769-vzd5b                   1/1       Running   0          29m
+
+HPAS:
+
+NAMESPACE   NAME       REFERENCE          TARGETS       MINPODS   MAXPODS   REPLICAS   AGE
+default     nginxext   Deployment/nginx   30/9 (avg)   1         3         3          29m
+
+```
+
+Voila
+
+### Troubleshooting
+
+- Make sure you have the Aggregation layer and the certificates set up as per the requirements section.
+- Always make sure the metrics you want to autoscale on are available
+As you create the HPA, the DCA will parse the manifest and query Datadog to try to fetch the metric.
```

Comment:
r/`As you create the HPA, the DCA will parse the manifest`/`As you create the HPA, the DCA parses the manifest`